src/util.rs

```diff
@@ -0,0 +1,26 @@
+use core::fmt::{self, Debug};
+
+pub struct Unshared<T> {
+    inner: T,
+}
+
+impl<T> Unshared<T> {
+    pub fn new(inner: T) -> Self {
+        Unshared { inner }
+    }
+
+    pub fn get_mut(&mut self) -> &mut T {
+        &mut self.inner
+    }
+}
+
+// Safety: we do not access `self.inner` in any `&self` method, so there is no way to have shared
+// references to it in multiple threads.
```

Comment:
This is incorrect with scoped threads, you could do something like

```rust
let mut foo = Unshared::new(...);
let bar = &*foo.get_mut();
spawn(move || { dbg!(bar) });
spawn(move || { dbg!(bar) });
```

Reply:
Oh, actually no, I'm wrong. Once you go to `&T` that is `!Send` because `T: !Sync`. So I think this is fine.